### PR TITLE
fix CTID in tx command returns invalidParams on lowercase hex

### LINKED ISSUE #4776: CTID in tx command returns invalidParams on lowercase hex
## Issue Description
The `ctid` field of the `tx` command, added in #4418, does not work if the given CTID contains lowercase hex.

## Steps to Reproduce
Send [this request](https://xrpl.org/websocket-api-tool.html?server=wss%3A%2F%2Fxrplcluster.com%2F&req=%7B%22id%22%3A%22example%20lowercase%20in%20CTID%22%2C%22command%22%3A%22tx%22%2C%22ctid%22%3A%22C005523e00000000%22%7D) to a rippled 1.12 server.

```json
{
  "id": "example lowercase in CTID",
  "command": "tx",
  "ctid": "C005523e00000000"
}
```

## Expected Result
Request should successfully find transaction E08D6E9754025BA2534A78707605E0601F03ACE063687A0CA1BDDACFCD1698C7 like [this example](https://xrpl.org/websocket-api-tool.html?server=wss%3A%2F%2Fxrplcluster.com%2F&req=%7B%22id%22%3A%22example%20working%20CTID%22%2C%22command%22%3A%22tx%22%2C%22ctid%22%3A%22C005523E00000000%22%7D). Aside from the ID, the difference is the `e` vs `E` in the CTID.

## Actual Result
Response is an `invalidParams` error message.

```json
{
  "error": "invalidParams",
  "error_code": 31,
  "error_message": "Invalid parameters.",
  "id": "example lowercase CTID",
  "request": {
    "command": "tx",
    "ctid": "C005523e00000000",
    "id": "example lowercase CTID"
  },
  "status": "error",
  "type": "response"
}
```

## Environment
rippled 1.12.0 via xrplcluster.com

### PR DESCRIPTION
**Pull request by yinyiqian1**
fix #4776 

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

**Review by shawnxie999 — APPROVED** [review]
looks good

**Review by scottschurr — APPROVED** [review]
Good fix, and thanks for including a test that exercises the new case.

I left some comments about things that, in my opinion, would improve the unit test.  But they are at your discretion.

File: src/test/rpc/Transaction_test.cpp
Comment (yinyiqian1) [reply src/test/rpc/Transaction_test.cpp:682]: Yes. It makes sense

**Review by scottschurr — APPROVED** [review]
I found a couple more nits in the most recent changes to the test.  But, again, nothing that should hold up merging.

File: src/test/rpc/Transaction_test.cpp
Comment (yinyiqian1) [reply src/test/rpc/Transaction_test.cpp:689]: yeah, sorry I missed it

**Review by scottschurr — APPROVED** [review]
:+1:

**Review by shawnxie999 — APPROVED** [review]

**Comment by scottschurr** [comment]
@yinyiqian1, this pull request has two approvals.  Are you ready for the pull request to be marked as passed?  Let me know.  Thanks.

**Comment by yinyiqian1** [comment]
> @yinyiqian1, this pull request has two approvals. Are you ready for the pull request to be marked as passed? Let me know. Thanks.

@scottschurr Yes. It can be marked as passed. Thank you Scott.

**Pull request merged**
